scripts/dockerfiles/Dockerfile.build

```diff
@@ -28,6 +28,7 @@ RUN yum install -y  \
       pkgconfig \
       systemd-devel \
       zlib-devel \
+      valgrind-devel \
```

Comment:
our main build dockerfile should not need to include valgrind... am I misunderstanding this?

Reply:
Does this picture help:
```
build
    ├── main-debug-base
    │   ├── 1.\ rebase-to-amazonlinux
    │   ├── init-debug-base
    │   │   ├── 1.\ dep-build-init
    │   │   ├── init-debug-fs
    │   │   └── init-debug-s3
    │   ├── main-debug-fs
    │   ├── main-debug-s3
    │   └── main-debug-valgrind
    └── main-release
        ├── 1.\ rebase-to-amazonlinux
        └── init-release
            └── 1.\ dep-build-init
```

Reply:
Actually, if DFLB_VALGRIND compiler flag is set you may need valgrind-devel as a dep.

https://github.com/aws/aws-for-fluent-bit/blob/mainline/Dockerfile.debug-base#L31

So you do need valgrind-devel

The deps in the build image are not actually included in any of the final images

Reply:
But wait. For non-debug build, should we enable -DFLB_VALGRIND? I think no. 

For the debug images, I think we need valgrind-devel at runtime for valgrind to work...

Reply:
https://github.com/aws/aws-for-fluent-bit/blob/mainline/Dockerfile.debug-base#L31

Currently it is featured in the debug images:
https://github.com/aws/aws-for-fluent-bit/blob/mainline/Dockerfile.debug-base#L31